rest-catalog/quarkus-rest-catalog/src/main/java/org/projectnessie/restcatalog/server/json/IcebergObjectMapperProducer.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.restcatalog.server.json;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Singleton;
+import org.projectnessie.restcatalog.util.Json;
+
+@ApplicationScoped
+public class IcebergObjectMapperProducer {
+
+  /**
+   * Produces the {@link ObjectMapper} used by the Iceberg REST API.
+   *
+   * <p>We already produce this specific mapper through {@link
+   * org.projectnessie.restcatalog.ee.javax.ContextObjectMapper}. But unfortunately,
+   * ContextResolver @Provider-annotated classes are a JAX-RS mechanism, and sometimes they are not
+   * picked up by the CDI container, for example in {@link
+   * io.quarkus.resteasy.reactive.jackson.runtime.serialisers.ServerJacksonMessageBodyReader}. So we
+   * also produce the mapper here, to make sure it is available everywhere.
+   *
+   * <p>The bean produced here will override the default bean produced by {@link
+   * io.quarkus.jackson.runtime.ObjectMapperProducer}.
+   */
+  @Singleton
+  public ObjectMapper produceObjectMapper() {
```

Comment:
I am not sure if this has become necessary because of the switch to resteasy-reactive, or if things were already failing before. But without this bean, the default ObjectMapper would be used to deserialize requests and the objects would be empty.

Reply:
This works for me